add fake fp16 net transforms

hyuen opened the pull request:
Differential Revision: D23080881

yinghai commented on caffe2/python/test/fakefp16_transform_test.py:
```diff
@@ -0,0 +1,23 @@
+        net_swish = core.Net("test_swish")
+        net_swish_init = core.Net("test_swish_init")
+
+        deq = core.CreateOperator("Int8DequantizeNNPI", ["Xq"], ["X"])
```
Nit I guess you can just do core.Int8DequantizeNNPI(...)

Commit:
add fake fp16 fusions to net transforms (#42927)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/42927

added fp16 fusion to net transforms
refactored the transforms as well as glow_transform to get out of opt/custom so that the OSS builds passed

Test Plan: added net runner tests for this

Differential Revision: D23080881

fbshipit-source-id: 61778dbf98e575fc5ce4e1e201121f5d199e0e4e
